contract/src/main/java/io/neow3j/contract/Nep5Token.java

```diff
@@ -161,4 +163,14 @@ public String transfer(Wallet wallet, ScriptHash to, BigDecimal amount)
                 .send();
     }
 
+    public String transferFromURI(Wallet wallet, NeoURI neoURI) throws IOException, ErrorResponseException {
+        ScriptHash to = neoURI.getAddress(); // TODO: 22.04.20 Michael: handle insufficient uri
+        BigDecimal amount = neoURI.getAmount();
+
+        if (this.scriptHash != neoURI.getAsset()) {
```

Comment:
I don't get what's the logic here... can you explain a bit?

Reply:
In the [Nep9](https://github.com/neo-project/proposals/blob/master/nep-9.mediawiki) proposal, a valid nep-5 smart contract script hash is required in the uri for smart contract invocations. This should be a check that the NeoURI's contract scripthash is equal to the Nep5Token instance.

Reply:
In Java, a safe comparison of string is with:
`.equals()`
If that's the case, please change it. Just check, as well, if the string you want to compare is not null.